### PR TITLE
fix(dialect-clickhouse): Support DROP statements

### PR DESCRIPTION
**Pull request by germainlefebvre4**
### Brief summary of the change made

Features:
* Support `DROP` statements:
  * `DROP DATABASE`
  * `DROP TABLE`
  * `DROP DICTIONARY`
  * `DROP USER`
  * `DROP ROLE`
  * `DROP QUOTA`
  * `DROP SETTINGS PROFILE`
  * `DROP VIEW`
  * `DROP FUNCTION`

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - ~`.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.~
  - [x] `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - [ ] Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- ~Added appropriate documentation for the change.~
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_clickhouse.py [hunk 1]
```diff
@@ -589,6 +589,224 @@ class CreateMaterializedViewStatementSegment(BaseSegment):
...
+
+    type = "drop_table_statement"
+
+    match_grammar = Sequence(
+        "DROP",
+        Ref.keyword("TEMPORARY", optional=True),
+        "TABLE",
+        Ref("IfExistsGrammar", optional=True),
+        Ref("TableReferenceSegment"),
+        # Ref("OnClusterClauseSegment", optional=True),
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_clickhouse.py:607]: What are all of these commented out OnClusterClauseSegments for?

Comment (germainlefebvre4) [reply src/sqlfluff/dialects/dialect_clickhouse.py:607]: It is to mind to change the ON CLUSTER block with a dedicated function 
I have 3 PRs that will refer to the future function 

do you prefer to remove it before merging? and fix this later?

Comment (WittierDinosaur) [reply src/sqlfluff/dialects/dialect_clickhouse.py:607]: If you're gonna follow up with this soon, I'm happy to leave it in 👍

**Review by WittierDinosaur — APPROVED** [review]
LGTM

**Pull request merged**